src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -186,7 +192,7 @@ class RegularSync(
       log.info("Received {} block headers in {} ms", headers.size, timeTaken)
       waitingForActor = None
       if (resolvingBranches) handleBlockBranchResolution(peer, headers.reverse)
```

Comment:
Somehow `headersQueue` can be empty here... resulting in `java.util.NoSuchElementException: head of empty list` 🤔

Reply:
Cause found, fix will be pushed soon

Reply:
✅